packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverter.kt

```diff
@@ -54,6 +54,23 @@ object JSTypeConverter {
     }
   }
 
+  /**
+   * Tests that a converted value can be added to a map for inclusion in the payload of a JS event
+   * without throwing
+   */
+  @JvmStatic
+  fun valueIsConvertibleToJSValue(value: Any?, containerProvider: ContainerProvider = DefaultContainerProvider): Boolean {
+    val testMap = containerProvider.createMap()
+    try {
+      val convertedValue = JSTypeConverter.convertToJSValue(value, containerProvider)
+      testMap.putGeneric("testKey", convertedValue)
+    } catch (e: Throwable) {
+      return false
+    }
+    return true
+  }
```

Comment:
I don't like this approach, but I understand the current limitation. We can't fix all the problems correctly because we would have to stop using `folly::dynamic` entirely. We can temporarily add such a function, but let's move it to the `expo-notifications` package.

Reply:
Sadly, `expo-notifications` can't access the `WritableMap` class, since it doesn't have a direct dependency on RN. Also, keeping this method here might help to keep it in sync if `JSTypeConverter` changes in future.

Reply:
This version no longer uses `WritableMap`, and I don't see how keeping it in `expo-modules-core` would help in syncing it with `JSTypeConverter`. You're using `JSTypeConverter.convertToJSValue` directly, which should be enough.

Reply:
I don't have a strong opinion of where this should live. If moving it to `expo-notifications` is easy then I guess that's fine.

If it is in `expo-modules-core` though, we should probably add a test for it. In `expo-notifications` it'd still be good to add a test but not as strictly necessary.

Reply:
@lukmccall the `convertToJSValue` does not throw for generic values. Placing that value inside a bundle and calling `convertToJSValue` is what causes the `WritableMap` exception.

Reply:
@wschurman @lukmccall I've changed the `filteredBundleForJSTypeConverter` implementation so that no `expo-modules-core` change is needed. The code is a little more complex, but works correctly in local testing. Also structured the implementation so that the check for what values may be converted is a replaceable function --  the idea is to make unit testing easier.